File: server/shoppe/jv-service/src/main/java/com/lcaohoanq/jvservice/domain/token/ITokenService.java
package com.lcaohoanq.jvservice.domain.token;

import com.lcaohoanq.jvservice.base.exception.DataNotFoundException;
import com.lcaohoanq.jvservice.domain.user.User;
import org.springframework.stereotype.Service;

@Service

public interface ITokenService {
    Token addToken(long userId, String token, boolean isMobileDevice);
    Token refreshToken(String refreshToken, User user) throws Exception;
    void deleteToken(String token, User user) throws DataNotFoundException;
    Token findUserByToken(String token) throws DataNotFoundException;
}

File: spcserver/kotlin/src/main/kotlin/com/lcaohoanq/kotlinbasics/domain/token/ITokenService.kt
package com.lcaohoanq.kotlinbasics.domain.token

import com.lcaohoanq.kotlinbasics.domain.user.User


interface ITokenService {

    fun addToken(userId: Long, token: String): Token
    fun refreshToken(refreshToken: String, user: User): Token
    fun deleteToken(token: String, user: User)
    fun findUserByToken(token: String): Token

}